res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -0,0 +1,2696 @@
+                && engine.getValue(S4Mk3.rightDeck.group, "play")) {
+                velocityRight = wheelAbsoluteMax * engine.getValue(S4Mk3.rightDeck.group, "rate_ratio") + rightMinRate;
+            }
+
+            // byte 2 > 127 rotates backward
+            const motor = [1, 32, 1, velocityLeft & (2 ** 8 - 1), velocityLeft >> 8,
+                1, 32, 1, velocityRight & (2 ** 8 - 1), velocityRight >> 8];
+            controller.send(motor, null, 49, true);
+        });
+    }
+    incomingData(data) {
+        const reportId = data[0];
+        if (reportId === 1) {
+            this.inPackets[1].handleInput(data.buffer);
+        } else if (reportId === 2) {
+            this.inPackets[2].handleInput(data.buffer);
+            // The master volume, booth volume, headphone mix, and headphone volume knobs
+            // control the controller's audio interface in hardware, so they are not mapped.
+        } else if (reportId === 3) {
+            // The 32 bit unsigned ints at bytes 8 and 36 always have exactly the same value,
+            // so only process one of them. This must be processed before the wheel positions.
+            const oldWheelTimer = wheelTimer;
+            const view = new DataView(data.buffer);
+            wheelTimer = view.getUint32(8, true);
+            // Processing first value; no previous value to compare with.
+            if (oldWheelTimer === null) {
+                return;
+            }
+            wheelTimerDelta = wheelTimer - oldWheelTimer;
+            if (wheelTimerDelta < 0) {
+                wheelTimerDelta += wheelTimerMax;
+            }
+
+            this.leftDeck.wheelRelative.input(view.getUint16(12, true));
+            this.rightDeck.wheelRelative.input(view.getUint16(40, true));
+        }
+    }
+    init() {
+        // sending these magic packets is required for the jog wheel LEDs to work
+        const wheelLEDinitPacket = Array(26).fill(0);
+        wheelLEDinitPacket[1] = 1;
+        wheelLEDinitPacket[2] = 3;
+        controller.send(wheelLEDinitPacket, null, 48);
+        wheelLEDinitPacket[0] = 1;
+        // hack around https://github.com/mixxxdj/mixxx/issues/10828
+        engine.beginTimer(35, () => { controller.send(wheelLEDinitPacket, null, 48); }, true);
+
+        // Init wheel timer data
+        wheelTimer = null;
+        wheelTimerDelta = 0;
+
+        // get state of knobs and faders
+        this.incomingData(new Uint8Array(controller.getInputReport(2)));
+    }
+    shutdown() {
+        // button LEDs
+        controller.send(new Array(94).fill(0), null, 128);
+
+        // meter LEDs
+        controller.send(new Array(78).fill(0), null, 129);
+
+        const wheelOutput = Array(40).fill(0);
+        // left wheel LEDs
+        // FIXME this data gets ignored due to https://github.com/mixxxdj/mixxx/issues/10828
+        // Unfortunately, because this is the teardown function, we cannot use a timer to delay the send
```

Comment:
Ooof, caught between two bugs :see_no_evil: I tried getting timers working on shutdown years ago but wasn't successful. There are some MIDI controllers out there also affected by this because they won't respond to sending tons of MIDI messages sent at once to turn off every LED.

Reply:
Yes, it is a very annoying bug to workaround. I hope this gets fixed soon but I thing @JoergAtGithub mentioned he was close to something for it so hopefully this "FIXME" will go away very soon :crossed_fingers: